pkgs/servers/http/nginx/modules.nix

```diff
@@ -136,4 +136,37 @@
       sha256 = "0h98a8kiw2qkqfavysm1v16kf4cs4h39j583wapif4p0qx3bbm89";
     };
   };
+
+  # PageSpeed module: https://developers.google.com/speed/pagespeed/module/
+  pagespeed = let
+    # https://developers.google.com/speed/pagespeed/module/build_ngx_pagespeed_from_source
+    # WARNING: This only works with Linux b/c the pre-built PSOL binary is only supplied for Linux.
+    # TODO: Build PSOL from source to support more platforms.
+
+    version = "1.11.33.4";
+
+    moduleSrc = fetchFromGitHub {
+      owner  = "pagespeed";
+      repo   = "ngx_pagespeed";
+      rev    = "v${version}-beta";
+      sha256 = "03dvzf1lgsjxcs1jjxq95n2rhgq0wy0f9ahvgascy0fak7qx4xj9";
+    };
+
+    psol = pkgs.fetchzip {
+      url    = "https://dl.google.com/dl/page-speed/psol/${version}.tar.gz";
```

Comment:
In fact, I think this same file would be used in the Apache `httpd` module, so we likely *should* move this somewhere higher in the architecture.